support of multimeasure for MQTT and AMQP transport

Chandradeep-NEC opened the pull request:
Multi-measure support to IoT Agent JSON in case of MQTT and AMQP transport (#462 )

Chandradeep-NEC commented:
HI @fgalan 

This is a gentle reminder!!
If you get time then please have a look on PR #498 . If the PR seems OK then please merge it in master.

fgalan commented on CHANGES_NEXT_RELEASE:
```diff
@@ -1 +1,2 @@
 Update Docker security practices (Add HEALTHCHECK, Use Anonymous User, Use two-stage build)
+Add: Support of multimeasure for MQTT and AMQP transport (#462)
```
What about multimeasure functionality for MQTT and AMQP in IOTA-UL (https://github.com/telefonicaid/iotagent-ul/)?

My guess is that probably in that agent it's already implemented, but it would be great to have a confirmation in positive o negative sense.

Chandradeep-NEC replied:
@fgalan 
I've verified the Multi measure functionality for MQTT and AMQP in IOTA-UL. It is working fine for both transports as well.
I've also added UT to check same functionality regarding AMQP. UT for MQTT is already written, i just executed and confirmed the successful execution.

fgalan replied:
Ok. Great!

Thus, it is positively confirmed by your analysis and this comment thread is NTC.

fgalan commented:
Looking to parent issue:

> Multimeasures has been implemented for HTTP transport. What is pending is to implement multimeasures for MQTT and AMPQ transports (and modify the "IMPORTANT NOTE" mark in documentation accordingly)

This PR should include also that modification to "IMPORTANT NOTE" in documentation.

Chandradeep-NEC commented:
> Looking to parent issue:
> 
> > Multimeasures has been implemented for HTTP transport. What is pending is to implement multimeasures for MQTT and AMPQ transports (and modify the "IMPORTANT NOTE" mark in documentation accordingly)
> 
> This PR should include also that modification to "IMPORTANT NOTE" in documentation.

I guess, you are talking about "IMPORTANT NOTE" added in usermanual.md file in IOTA-JSON regarding not support of multiple measure functionality for MQTT and AMQP transport. 
I suggest that this note should be removed. Please confirm.

fgalan commented:
> 
> 
> > Looking to parent issue:
> > > Multimeasures has been implemented for HTTP transport. What is pending is to implement multimeasures for MQTT and AMPQ transports (and modify the "IMPORTANT NOTE" mark in documentation accordingly)
> > 
> > 
> > This PR should include also that modification to "IMPORTANT NOTE" in documentation.
> 
> I guess, you are talking about "IMPORTANT NOTE" added in usermanual.md file in IOTA-JSON regarding not support of multiple measure functionality for MQTT and AMQP transport.
> I suggest that this note should be removed. Please confirm.

Yes, I mean this note:

> IMPORTANT NOTE: at the present moment, multimeasures only work for HTTP. Support for other transports is still pending (see issue #391).

I agree: it has to be removed, given that as far as I understand this PR completed the multimeasure support for all the supported transport protocols.

fgalan commented:
> I agree: it has to be removed, given that as far as I understand this PR completed the multimeasure support for all the supported transport protocols.

In fact, looking to the current status of the PR it seems it is already removed. Is that correct?

Chandradeep-NEC commented:
> > I agree: it has to be removed, given that as far as I understand this PR completed the multimeasure support for all the supported transport protocols.
> 
> In fact, looking to the current status of the PR it seems it is already removed. Is that correct?

Yes, I've removed and added the modification in PR

fgalan commented:
It seems there is some overlapping between this PR and PR #410 in docs/usermanual.md file. Thus, I'd suggest to merge first PR #410 (older, singler and almost finished), then upgrade this PR with master and review and merge this PR.

What do you think?

Chandradeep-NEC commented:
> It seems there is some overlapping between this PR and PR #410 in docs/usermanual.md file. Thus, I'd suggest to merge first PR #410 (older, singler and almost finished), then upgrade this PR with master and review and merge this PR.
> 
> What do you think?

Yes it is OK to merge PR #410  first and then merge #498 . But if you merge #498  first then still there will be no problem, i'll re-base for PR #410. For PR #498, Only one statement has been removed from usermanual.md file It is good to merge this PR first.

Chandradeep-NEC commented:
@fgalan 
Its long pending PR. Request you please merge the PR

Chandradeep-NEC commented:
@fgalan 
Is there any review pending?

fgalan commented:
> 
> 
> > It seems there is some overlapping between this PR and PR #410 in docs/usermanual.md file. Thus, I'd suggest to merge first PR #410 (older, singler and almost finished), then upgrade this PR with master and review and merge this PR.
> > What do you think?
> 
> Yes it is OK to merge PR #410 first and then merge #498 . But if you merge #498 first then still there will be no problem, i'll re-base for PR #410. For PR #498, Only one statement has been removed from usermanual.md file It is good to merge this PR first.

Now that PR #410 has been merged, I think this PR is ready to be synced with master. A small conflict in CHANGEST_NEXT_RELEASE need to be solved in that process. Thanks!

Chandradeep-NEC commented:
@fgalan 
Ok. I'm trying to resolve the conflict

Chandradeep-NEC commented:
conflict resolved..

Chandradeep-NEC commented:
@fgalan 
Gentle reminder!!
Request you to please check the PR. conflicts are resolved. It can be merged now

fgalan reviewed (approved):
LGTM
